android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java

```diff
@@ -1255,19 +1255,22 @@ private void showNTPImage(NTPImage ntpImage) {
 
         boolean wasWallpaperShown = true;
         if (ntpImage instanceof Wallpaper && ((Wallpaper) ntpImage).isRichMedia()) {
-            setupSponsoredBackgroundContent();
-        } else if (ntpImage instanceof Wallpaper
-                && NTPImageUtil.isReferralEnabled()
-                && Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-            setBackgroundImage(ntpImage);
-
-        } else if (UserPrefs.get(ProfileManager.getLastUsedRegularProfile())
-                        .getBoolean(BravePref.NEW_TAB_PAGE_SHOW_BACKGROUND_IMAGE)
-                && mSponsoredTab != null
-                && NTPImageUtil.shouldEnableNTPFeature()) {
-            setBackgroundImage(ntpImage);
+            setupSponsoredBackgroundContent((Wallpaper) ntpImage);
         } else {
-            wasWallpaperShown = false;
+            maybeResetSponsoredRichMediaBackground();
+
+            if (ntpImage instanceof Wallpaper
+                    && NTPImageUtil.isReferralEnabled()
+                    && Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```

Comment:
we don't support `Build.VERSION_CODES.M` and lower already, so could be removed.

Reply:
These checks were in code before my changes. I filed an issue to fix it in our codebase: https://github.com/brave/brave-browser/issues/49506